### PR TITLE
feat: text link codemod 

### PR DESCRIPTION
**Pull request by suevalov**
# Purpose of PR

Adds a codemod to migrate TextLink component to v4.
Adds reusable functions that can be used in other codemods.

### CODE REVIEW

File: packages/forma-36-codemod/transforms/v4-text-link.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+          j,
+          icons: usedIcons,
+        });
+      }
+
+      return modifiedAttributes;
+    },
+  });
+
+  source = changeImport(j, source, {
```
Comment (suevalov) [inline packages/forma-36-codemod/transforms/v4-text-link.js:75]: Probably sometimes we don't want a change of the import to happen... For example, in Compose codebase. It looks like we need to make it controllable.

File: packages/forma-36-codemod/transforms/v4-text-link.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    componentName,
+    from: '@contentful/forma-36-react-components',
+    to: '@contentful/f36-components',
+  });
+
+  source = addIconImports({ j, source, icons: usedIcons });
```
Comment (suevalov) [inline packages/forma-36-codemod/transforms/v4-text-link.js:81]: Looks like some icon components names can clash with Forma component names:

* Heading
* Asset

Should we add postfix all icon components with with `Icon`?

`Heading` => `HeadingIcon`
`Asset` => `AssetIcon`

File: packages/forma-36-codemod/utils/index.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+const { addProperty } = require('./addProperty');
+const { changeProperties } = require('./changeProperties');
+const { updatePropertyValue } = require('./updatePropertyValue');
+const { getProperty } = require('./getProperty');
+const { addImport } = require('./addImport');
+const { updateIcons, addIconImports } = require('./updateIcons');
+const { removeComponentImport } = require('./removeComponentImport');
+const { changeImport } = require('./changeImport');
+
+module.exports = {
```
Comment (burakukula) [inline packages/forma-36-codemod/utils/index.js:14]: Awesome! 👏

**Review by burakukula — APPROVED** [review]
🎉 
Only one thing: we might want to extend the README a bit to cover your changes.

File: packages/forma-36-codemod/transforms/v4-text-link.js (hunk 2 above)
Comment (burakukula) [reply packages/forma-36-codemod/transforms/v4-text-link.js:81]: yeah, I think it would cover any potential cases like that in the future.

**Review by massao — APPROVED** [review]
👏

File: packages/forma-36-codemod/utils/getProperty.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+  attributes.forEach((attribute) => {
+    if (found === null) {
+      if (attribute.name.name === propertyName) {
+        found = attribute;
+      }
+    }
+  });
+
+  return found;
```
Comment (bgutsol) [inline packages/forma-36-codemod/utils/getProperty.js:15]: Why not just:
```suggestion
  const found = attributes.find((attribute) => attribute.name.name === propertyName)

  return found || null;
```

File: packages/forma-36-codemod/transforms/v4-text-link.js (hunk 2 above)
Comment (bgutsol) [reply packages/forma-36-codemod/transforms/v4-text-link.js:81]: +1 to add postfix

**Review by bgutsol — COMMENTED** [review]
Great work with utils functions 👏
One important comment for the `TextLink` codemod is to add `as="button"` to the places where we use `TextLink` without `href` prop. 
[Before the default was `button` element when you not provide `href`](https://github.com/contentful/forma-36/pull/1125/files#diff-2ecb4e98406d62467375ddf356870f24199fcc08728d2eee4d83c9fd7b82c442L92)
[In v4 the default is `a` element even if the `href` is not provided](https://github.com/contentful/forma-36/pull/1125/files#diff-3bbbaafb6051437ff8b342074584b898af8f71a6401a8c28f34cc34bcff9ce4bR38)

I did it, because we added `as` prop and IMO it should be the only one prop that defines component element. But we can also revisit component implementation if it's not convenient for us.

File: packages/forma-36-codemod/utils/getProperty.js (hunk 4 above)
Comment (suevalov) [reply packages/forma-36-codemod/utils/getProperty.js:15]: cause Yoda I like :)

File: packages/forma-36-codemod/transforms/v4-text-link.js (hunk 2 above)
Comment (suevalov) [reply packages/forma-36-codemod/transforms/v4-text-link.js:81]: going to changes this in a subsequent PR

**Comment by suevalov** [comment]
> One important comment for the TextLink codemod is to add as="button" to the places where we use TextLink without href prop.

Good remark! Changed the implementation, but we will need to learn if it's a good default while we're adopting it.

**Review by bgutsol — APPROVED** [review]

**Pull request merged**